### PR TITLE
Milestone 2 - Connor OBrien

### PR DESCRIPTION
**Pull request by connorpobrien**
# Milestone: #2
# Developer: Connor O'Brien

### Pull Request Summary

Submission for milestone 2

### Pull Request Checklist
- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit message follows our guidelines
- [x] The milestone number is specified
- [x] The developer name is specified
- [x] The summary is completed

### Reviewer Checklist  - PULL REQUEST REVIEWER ONLY

IMPORTANT: The rest of the sections in this checklist should only be filled out by authorized pull request reviewers. If you are the individual template contributor, do not fill out the rest of the fields or check the boxes.

NOTE: Milestones can only be completed when all boxes are checked.

- [x] The code is fully reviewed
- [x] Meaningful feedback is given
- [x] Comment that you have reviewed the code
- [x] This box is checked

### CODE REVIEW

File: milestone2.ts [hunk 1]
```diff
@@ -83,34 +85,65 @@ const userData: Volunteer[] = [
...
+  let count = 0;
+  for (var volunteer of data) {
+    if (volunteer.city == "SF") {
+      return count;
+    }
+    else {
+      count += 1;
+    }
+  }
+  return -1;
```
Comment (vwinstea) [inline milestone2.ts]: For questions 2-4 please use the data handling functions shown [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array). For loops are good, but its important to be familiar with lambda function syntax too. Sorry if the instructions were unclear!

File: milestone2.ts [hunk 2]
```diff
@@ -125,37 +158,44 @@ let kyle: Volunteer = {
...
 // Question 7: Object Destructuring
 /* Now that we have our updated kyle, lets use object destructuring to get his name, age, and city
  */
 
 type GetVolunteerInfo = (vol: Volunteer) => String;
 let getVolunteerInfo: GetVolunteerInfo = (vol) => {
-  let name, age, city; // Code here
+  let name = vol.name;
+  let age = vol.age;
+  let city = vol.city;
```
Comment (vwinstea) [inline milestone2.ts]: Please use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring) for this question

**Review by vwinstea — CHANGES_REQUESTED** [review]
Great work! All of the tests pass, but before wrapping up this milestone, please update your code for questions 2-4 and 7 by following the comments below.

**Comment by connorpobrien** [comment]
changes made ✅

**Review by vwinstea — COMMENTED** [review]
Much better, thanks for making the fixes! Question 7 is still a bit off, you used array destructuring instead of [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring), but that was a pretty common mistake for this milestone. Nice work!

**Comment by vwinstea** [comment]
Milestone 2 complete!

**Pull request closed without merging**